pkg/kubelet/cri/streaming/portforward/httpstream.go

```diff
@@ -202,55 +206,61 @@ func (h *httpStreamHandler) requestID(stream httpstream.Stream) string {
 			requestID = strconv.Itoa(int(stream.Identifier()) - 2)
 		}
 
-		klog.V(5).InfoS("Connection automatically assigning request ID from stream type and stream ID", "connection", h.conn, "request", requestID, "streamType", streamType, "stream", stream.Identifier())
+		klog.V(5).InfoS("Connection automatically assigning request ID from stream type and stream ID", "request", requestID, "streamType", streamType, "stream", stream.Identifier())
 	}
 	return requestID
 }
 
 // run is the main loop for the httpStreamHandler. It processes new
 // streams, invoking portForward for each complete stream pair. The loop exits
 // when the httpstream.Connection is closed.
-func (h *httpStreamHandler) run() {
-	klog.V(5).InfoS("Connection waiting for port forward streams", "connection", h.conn)
+func (h *httpStreamHandler) run(ctx context.Context) {
+	klog.V(5).InfoS("Connection waiting for port forward streams")
 Loop:
 	for {
 		select {
+		case <-ctx.Done():
+			klog.V(5).InfoS("Connection context closed")
+			break Loop
 		case <-h.conn.CloseChan():
-			klog.V(5).InfoS("Connection upgraded connection closed", "connection", h.conn)
+			klog.V(5).InfoS("Connection upgraded connection closed")
 			break Loop
 		case stream := <-h.streamChan:
 			requestID := h.requestID(stream)
 			streamType := stream.Headers().Get(api.StreamType)
-			klog.V(5).InfoS("Connection request received new type of stream", "connection", h.conn, "request", requestID, "streamType", streamType)
+			klog.V(5).InfoS("Connection request received new type of stream", "request", requestID, "streamType", streamType)
 
 			p, created := h.getStreamPair(requestID)
 			if created {
-				go h.monitorStreamPair(p, time.After(h.streamCreationTimeout))
+				go func() {
+					timer := time.NewTimer(h.streamCreationTimeout)
+					defer timer.Stop()
```

Comment:
if we are plumbing a context, does it make sense to add the timeout to the context and avoid the timer?
```
ctx, cancel := context.WithTimeout(ctx, time.Duration(h.streamCreationTimeout))
defer cancel()
``

Reply:
Yeah I suppose that would make this a little cleaner.